Experimental/CameraComponent/FirstPersonCamera.cs

```diff
@@ -1,264 +1,166 @@
+                float curr;
+                _propertySet.TryGetScalar("Roll", out curr);
+                return curr;
             }
             set
             {
-                if (UseAnimations)
-                {
-                    propertySet.InsertScalar("Roll", value);
-                }
-                else
-                {
-                    roll = value;
-                    RaisePropertyChanged("Roll");
-                }
+                _propertySet.InsertScalar("Roll", value);
             }
         }
+
+        // Camera's location in world space
         public Vector3 Position
         {
             get
             {
-                if(UseAnimations)
-                {
-                    Vector3 curr = Vector3.Zero;
-                    return (propertySet.TryGetVector3("Position", out curr) == CompositionGetValueStatus.Succeeded) ? curr : position;
-                }
-                else
-                {
-                    return position;
-                }
+                Vector3 curr;
+                _propertySet.TryGetVector3("Position", out curr);
+                return curr;
             }
             set
             {
-                if (UseAnimations)
-                {
-                    propertySet.InsertVector3("Position", value);
-                }
-                else
-                {
-                    position = value;
-                    RaisePropertyChanged("Position");
-                }
+                _propertySet.InsertVector3("Position", value);
             }
         }
-        public Vector3 LookDirection
+
+        // The camera's projection, orthographic or perspective
+        public Projection Projection
         {
-            get => Vector3.Zero;
+            get => _projection;
             set
             {
-                Vector3 direction = value;
-                if (value != Vector3.Zero)
+                _projection = value;
+
+                // create view matrix based on the camera's rotation and position
+                var matPos = "Matrix4x4.CreateTranslation(-FPCamera.Position)";
+                var matRoll = "Matrix4x4.CreateFromAxisAngle(Vector3(0, 0, 1), -FPCamera.Roll)";
+                var matPitch = "Matrix4x4.CreateFromAxisAngle(Vector3(1, 0, 0), -FPCamera.Pitch)";
+                var matYaw = "Matrix4x4.CreateFromAxisAngle(Vector3(0, 1, 0), -FPCamera.Yaw)";
+                var viewMat = matPos + "*" + matYaw + "*" + matPitch + "*" + matRoll;
+
+                // create a matrix that is the product of the camera's view matrix and the projection's projection matrix
+                var modelViewProjMatExpression = _compositor.CreateExpressionAnimation();
+                if (_projection == null) // if null then the default is an orthographic projection
                 {
-                    direction = Vector3.Normalize(value);
+                    OrthographicProjection defaultProj = new OrthographicProjection();

```

Comment:
The scoping of defaultProj is such that it will be garbage collected instantly, and only its propertySet will be kept alive through the animation it is attached to.  Just making sure that's intentional to lose the Projection object.

Reply:
That was my intention. I thought it would be better to keep the reference as null but and have the animation follow the default behavior. Do you think it would be better to actually assign the Projection to something when the user gives it null?